crates/epaint/src/stroke.rs

```diff
@@ -52,3 +54,116 @@ impl std::hash::Hash for Stroke {
         color.hash(state);
     }
 }
+
+/// How paths will be colored.
+#[derive(Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub enum ColorMode {
+    /// The entire path is one solid color, this is the default.
+    Solid(Color32),
+
+    /// Provide a callback which takes in a UV coordinate and converts it to a color. The UVs are relative to the bounding box of the path.
+    /// If you need something more custom, consider [`ColorMode::UVBounds`]
+    ///
+    /// **This cannot be serialized**
+    #[cfg_attr(feature = "serde", serde(skip))]
+    UV(Arc<Box<dyn Fn(Pos2) -> Color32 + Send + Sync>>),
+
+    /// Provide a callback which takes in a UV coordinate and converts it to a color. The UVs are relative to the Rect provided.
```

Comment:
Again, this need more explanation. Isa the given `Pos2` just a coordinate within the `Rect`? So that `pos == rect.left_top()` for a vertex that starts in the left-top corner of the bounding box?